syntax = "proto3";
package csce438;
import "google/protobuf/timestamp.proto";

// ------------------------------------------------------------
// The coordinator service definition
// These are the different functionalities exposed by service
// ------------------------------------------------------------

//Init and Heartbeat potentially redundant
service CoordService{
  rpc Heartbeat (ServerInfo) returns (Confirmation) {}
  rpc GetServer (ID) returns (ServerInfo) {}
  // ZooKeeper API here
  rpc create (ServerInfo) returns (PathStatus) {}
  rpc exists (ServerInfo) returns (PathStatus) {}
}

//server info message definition
message ServerInfo{
  int32 serverID = 1;
  string hostname = 2;
  string port = 3;
  string type = 4;
  int32 clusterID = 5;
}

//confirmation message definition
message Confirmation{
  bool status = 1;
}

//id message definition
message ID{
  int32 id = 1;
}

//id message definition
message PathAndData{
  string path = 1;
  string hostname = 2;
  string port = 3;
}

//id message definition
message Path{
  string path = 1;
}

//id message definition
message PathStatus{
  bool status = 1;
}

